### PR TITLE
AIRFLOW-509 - Create operator to delete tables in BigQuery

### PR DESCRIPTION
**Pull request by illop**
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- https://issues.apache.org/jira/browse/AIRFLOW-509

Per Apache guidelines you need to create a [Jira issue](https://issues.apache.org/jira/browse/AIRFLOW/).

Testing Done:
- Unittests are required, if you do not include new unit tests please
  specify why you think this is not required. We like to improve our
  coverage so a non existing test is even a better reason to include one.

Reminders for contributors (REQUIRED!):
- Your PR's title must reference an issue on 
  [Airflow's JIRA](https://issues.apache.org/jira/browse/AIRFLOW/). 
  For example, a PR called "[AIRFLOW-1] My Amazing PR" would close JIRA 
  issue #1. Please open a new issue if required!
- Please squash your commits when possible and follow the [How to write a good git commit message](http://chris.beams.io/posts/git-commit/). 
  Summarized as follows:
  1. Separate subject from body with a blank line
  2. Limit the subject line to 50 characters
  3. Do not end the subject line with a period
  4. Use the imperative mood in the subject line (add, not adding)
  5. Wrap the body at 72 characters
  6. Use the body to explain what and why vs. how

### CODE REVIEW

**Comment by illop** [comment]
NOTE: This is still very much in progress. Dropping the PR in here so that I can see how the tests are looking as I work on this; will update and squash my commits once it's done.

**Comment by illop** [comment]
Status: Operator written, need to write a test DAG on my machine to confirm that it works as designed.

**Comment by illop** [comment]
This is ready to go!

File: airflow/contrib/hooks/bigquery_hook.py [hunk 1]
```diff
@@ -23,10 +23,12 @@
 
 import logging
 import time
+import httplib2
```
Comment (criccomini) [inline airflow/contrib/hooks/bigquery_hook.py]: This isn't used

File: airflow/contrib/operators/bigquery_operator.py [hunk 2]
```diff
@@ -76,3 +76,48 @@ def execute(self, context):
         cursor = conn.cursor()
         cursor.run_query(self.bql, self.destination_dataset_table, self.write_disposition,
                          self.allow_large_results, self.udf_config)
+
+class BigQueryDeleteOperator(BaseOperator):
```
Comment (criccomini) [inline airflow/contrib/operators/bigquery_operator.py]: Could you put this in its own class?

BigQueryTableDeleteOperator

and

bigquery_table_delete_operator.py

File: airflow/contrib/operators/bigquery_operator.py [hunk 3]
```diff
@@ -76,3 +76,48 @@ def execute(self, context):
...
+            will be deleted.
+        :type deletion_dataset_table: string
+        :param bigquery_conn_id: reference to a specific BigQuery hook.
+        :type bigquery_conn_id: string
+        :param delegate_to: The account to impersonate, if any.
+            For this to work, the service account making the request must have domain-wide
+            delegation enabled.
+        :type delegate_to: string
+        :param ignore_if_missing: if True, then return success even if the
+        requested table does not exist.
```
Comment (criccomini) [inline airflow/contrib/operators/bigquery_operator.py]: Nit: spacing. Should add more spaces to make it line up with "For this..." above

File: airflow/contrib/hooks/bigquery_hook.py [hunk 4]
```diff
@@ -476,6 +478,41 @@ def get_tabledata(self, dataset_id, table_id,
...
+        """
+        Delete an existing table from the dataset;
+        If the table does not exist, return an error unless ignore_if_missing
+        is set to True.
+        :param deletion_dataset_table: A dotted
+            (<project>.|<project>:)<dataset>.<table> that indicates which table
+            will be deleted.
+        :type deletion_dataset_table: str
+        :param ignore_if_missing: if True, then return success even if the
+        requested table does not exist.
```
Comment (criccomini) [inline airflow/contrib/hooks/bigquery_hook.py:489]: Nit: spacing

File: airflow/contrib/hooks/bigquery_hook.py [hunk 5]
```diff
@@ -476,6 +478,41 @@ def get_tabledata(self, dataset_id, table_id,
...
+        try:
+            tables_resource = self.service.tables() \
+                .delete(projectId=deletion_project,
+                        datasetId=deletion_dataset,
+                        tableId=deletion_table) \
+                .execute()
+            logging.info('deleted table %s:%s.%s.',
+                         deletion_project, deletion_dataset, deletion_table)
+        except HttpError:
+            if ignore_if_missing == False:
```
Comment (criccomini) [inline airflow/contrib/hooks/bigquery_hook.py]: Pythonic: `if not ignore_if_missing:`

**Review by criccomini — CHANGES_REQUESTED** [review]

File: airflow/contrib/hooks/bigquery_hook.py (hunk 1 above)
Comment (illop) [reply airflow/contrib/hooks/bigquery_hook.py]: LOL forgot to remove this.

**Comment by illop** [comment]
I think I've addressed all of the above. Please confirm and I'll squash down the latest commit.

File: airflow/contrib/hooks/bigquery_hook.py [hunk 6]
```diff
@@ -27,6 +27,7 @@
 from airflow.contrib.hooks.gcp_api_base_hook import GoogleCloudBaseHook
 from airflow.hooks.dbapi_hook import DbApiHook
 from apiclient.discovery import build
+from apiclient.discovery import HttpError
```
Comment (criccomini) [inline airflow/contrib/hooks/bigquery_hook.py]: This line can be combined with line above:

```
from apiclient.discovery import build, HttpError
```

**Review by criccomini — CHANGES_REQUESTED** [review]

**Pull request closed without merging**